runtime/src/snapshot_utils.rs

```diff
@@ -2299,7 +2486,78 @@ fn rebuild_bank_from_snapshots(
 
     bank.status_cache.write().unwrap().append(&slot_deltas);
 
-    info!("Loaded bank for slot: {}", bank.slot());
+    info!("Rebuilt bank for slot: {}", bank.slot());
+    Ok(bank)
+}
+
+#[allow(clippy::too_many_arguments)]
+fn rebuild_bank_from_snapshot(
+    bank_snapshot: &BankSnapshotInfo,
+    account_paths: &[PathBuf],
+    storage_and_next_append_vec_id: StorageAndNextAppendVecId,
+    genesis_config: &GenesisConfig,
+    runtime_config: &RuntimeConfig,
+    debug_keys: Option<Arc<HashSet<Pubkey>>>,
+    additional_builtins: Option<&Builtins>,
+    account_secondary_indexes: AccountSecondaryIndexes,
+    limit_load_slot_count_from_snapshot: Option<usize>,
+    shrink_ratio: AccountShrinkThreshold,
+    verify_index: bool,
+    accounts_db_config: Option<AccountsDbConfig>,
+    accounts_update_notifier: Option<AccountsUpdateNotifier>,
+    exit: &Arc<AtomicBool>,
+) -> Result<Bank> {
+    info!(
+        "Rebuilding bank from snapshot {}",
+        bank_snapshot.snapshot_dir.display(),
+    );
+
+    let snapshot_root_paths = SnapshotRootPaths {
+        full_snapshot_root_file_path: bank_snapshot.snapshot_path(),
+        incremental_snapshot_root_file_path: None,
+    };
+
+    let bank = deserialize_snapshot_data_files(&snapshot_root_paths, |snapshot_streams| {
+        Ok(bank_from_streams(
+            SerdeStyle::Newer,
+            snapshot_streams,
+            account_paths,
+            storage_and_next_append_vec_id,
+            genesis_config,
+            runtime_config,
+            debug_keys,
+            additional_builtins,
+            account_secondary_indexes,
+            limit_load_slot_count_from_snapshot,
+            shrink_ratio,
+            verify_index,
+            accounts_db_config,
+            accounts_update_notifier,
+            exit,
+        )?)
+    })?;
+
+    let status_cache_path = bank_snapshot
+        .snapshot_dir
+        .join(SNAPSHOT_STATUS_CACHE_FILENAME);
+    let slot_deltas = deserialize_snapshot_data_file(&status_cache_path, |stream| {
+        info!(
+            "Rebuilding status cache from {}",
+            status_cache_path.display()
+        );
+        let slot_deltas: Vec<BankSlotDelta> = bincode::options()
+            .with_limit(MAX_SNAPSHOT_DATA_FILE_SIZE)
+            .with_fixint_encoding()
+            .allow_trailing_bytes()
+            .deserialize_from(stream)?;
```

Comment:
this feels like it's pulled/copied from somewhere else and we should probably wrap the deserialization in a function instead of duplicating the logic

Reply:
It is copied from the function immediately above -- fn rebuild_bank_from_unarchived_snapshots.

I think in the future we could always rebuild bank from a snapshot dir, so the functions building bank directly from archive could be removed.  In that case, they will be no duplication.

Reply:
sure, but we haven't removed it yet so we do have duplicated code 😄 

if we refactor or change `slot_delta` (de)serialization before remove above function, then we have a higher chance of missing one of these than if we have a common function for doing it.

Reply:
This function just does one thing -- calling another function with the deserializer closure.  Should the closure be factor out instead to be a named function?

Reply:
Added fn deserialize_status_cache